### PR TITLE
Bugfix: Form Serialization Crash

### PR DESCRIPTION
**Pull request by ctsims**
Fixes an issue where when our caching optimization fails and it crashes the app

http://manage.dimagi.com/default.asp?161629#898010

### CODE REVIEW

File: app/src/org/odk/collect/android/tasks/FormLoaderTask.java [hunk 1]
```diff
@@ -329,28 +343,34 @@ public FormDef deserializeFormDef(File formDef) {
...
-                e.printStackTrace();
-            } catch (IOException e) {
-                e.printStackTrace();
+            } finally {
+                //make sure we clean up the stream
+                if(fos != null) {
+                    try {
+                        fos.close();
+                    } catch (IOException e) {
+                        // there was already an error, this one is just cruft
```
Comment (phillipm) [inline app/src/org/odk/collect/android/tasks/FormLoaderTask.java]: Not sure I follow the logic of this try/catch
If we are already not catching IOException and letting it bubble up, why wrap fos.close() in a try? Wouldn't it suffice to just close the stream and let any exceptions propagate up the call stack?

Comment (ctsims) [inline app/src/org/odk/collect/android/tasks/FormLoaderTask.java]: In this case we don't want to trump the exception getting thrown from fd.writeExternal. If, for instance, we threw a different type of exception (in the ticket it's a stack overflow), and then while cleaning up fos.close() errored out, we'll end up logging/reporting fos.close()'s exception path. 

It did just occur to me, however, that I shouldn't be closing dos above if we're closing fos in the finally cleanup, either way, I'll fix that

File: app/src/org/odk/collect/android/tasks/FormLoaderTask.java [hunk 2]
```diff
@@ -329,28 +344,38 @@ public FormDef deserializeFormDef(File formDef) {
...
+            } finally {
+                //make sure we clean up the stream
+                if(outputStream != null) {
+                    try {
+                        outputStream.close();
+                    } catch (IOException e){
+                        // Swallow this. If we threw an exception from inside the 
+                        // try, this close exception will trump it on the return 
+                        // path, and we care a lot more about that exception
+                        // than this one.
```
Comment (ctsims) [inline app/src/org/odk/collect/android/tasks/FormLoaderTask.java]: @phillipm Let me know if this clarifies it

**Comment by phillipm** [comment]
@ctsims yeah, that makes sense. Thanks for the clarification!

**Pull request merged**